src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Serialization.cs

```diff
@@ -181,26 +181,49 @@ public void WriteTo(ObjectWriter writer)
                     }
                 }
             }
+
+            // sortedNodes is an array of Node instances which is often sorted by Node.Name by the caller. This method
```

Comment:
often?

Reply:
It's certainly not a precondition, and the algorithm will work correctly whether or not it's sorted.

Reply:
when does it not happen?

this is one of those cases where i would prefer our code be stricter and precise about how it expects things ot be working when dealing entirely with internal data.  If we always ensure this is sorted, we can state as such and throw if not.  if we don't always ensure this, we can doc when this isn't the case.  Such documentation serves to better explain our own beliefs aobut the system for the future, as well as more easily catch when our expectations are potentially broken in the future.